assets/js/components/Dashboard/NewPaMessage/StationsAndZones/SelectZonesPage.tsx

```diff
@@ -0,0 +1,541 @@
+    };
+  }, [selectedRouteFilter]);
+
+  useEffect(() => {
+    const routes = fp.flow(
+      fp.flatMap((place: Place) =>
+        place.screens.filter((s) => value.includes(s.id)),
+      ),
+      fp.flatMap((screen: Screen) => screen.route_ids),
+      fp.uniq,
+      fp.groupBy((routeID: string) => {
+        if (routeID.startsWith("Green")) {
+          return "Green";
+        }
+
+        if (routeToRouteIDMap["Bus"].includes(routeID)) {
+          return "Bus";
+        }
+
+        if (routeToRouteIDMap["Silver"].includes(routeID)) {
+          return "Silver";
+        }
+
+        return routeID;
+      }),
+    )(places);
+
+    setSelectedRoutes(routes);
+    setSelectedRouteFilter(Object.keys(routes)[0]);
+    setNewPlaces(
+      places.filter((p) =>
+        places
+          .filter((p) => p.screens.some((s) => value.includes(s.id)))
+          .map((p) => p.id)
+          .includes(p.id),
+      ),
+    );
+  }, [places]);
+
+  const filteredPlaces = useMemo(() => {
+    return newPlaces.filter((p) =>
+      p.screens.some((s) =>
+        s.route_ids?.some((r) =>
+          routeToRouteIDMap[selectedRouteFilter]?.some((a) => r === a),
+        ),
+      ),
+    );
+  }, [selectedRouteFilter]);
+
+  const allScreens = useMemo(
+    () =>
+      filteredPlaces.flatMap((p) => {
+        return p.screens.filter(
+          (s) =>
+            fp.intersection(routeToRouteIDMap[selectedRouteFilter], s.route_ids)
+              .length > 0,
+        );
+      }),
+    [filteredPlaces],
+  );
+
+  const screensByZone = useMemo(() => {
+    const screensWithAZone = allScreens.filter((s) => s.zone);
+    return fp.groupBy((s) => s.zone, screensWithAZone);
+  }, [allScreens]);
+
+  const areAllGroupsSelected = (groups: Array<ReadonlyArray<string>>) =>
+    groups.every((group) =>
+      group.every((zone) => {
+        return signIDs(screensByZone[zone] ?? []).every(isSelected);
+      }),
+    );
+
+  const selectGroups = (groups: Array<ReadonlyArray<string>>) => {
+    const signsWithoutSelectedRoute = fp.without(signIDs(allScreens), value);
+
+    const newScreens = groups.flatMap((group) =>
+      group.flatMap((zone) => signIDs(screensByZone[zone] ?? [])),
+    );
+
+    onChange(fp.uniq([...signsWithoutSelectedRoute, ...newScreens]));
+  };
+
+  const isAllSelected =
+    value.length > 0 &&
+    areAllGroupsSelected([LEFT_ZONES, MIDDLE_ZONES, RIGHT_ZONES]);
+  const isLeftSelected =
+    !isAllSelected && areAllGroupsSelected([LEFT_ZONES, MIDDLE_ZONES]);
+  const isRightSelected =
+    !isAllSelected && areAllGroupsSelected([MIDDLE_ZONES, RIGHT_ZONES]);
+
+  const hasRailFilter = () =>
+    [
+      "Green",
+      "Green-B",
+      "Green-C",
+      "Green-D",
+      "Green-E",
+      "Blue",
+      "Red",
+      "Orange",
+      "Mattapan",
+    ].includes(selectedRouteFilter);
+
+  const massSelectClassName = (selected: boolean) =>
+    selected ? "button-primary" : "button-secondary-outline";
```

Comment:
**quibble:** i would prefer something that acted as an `active` class on a button that only included the properties required to make the given button appear active instead of having to swap out two different kinds of button classes.

meaning you could have a class list like `button-primary` and when it was active you would have `button-primary button-active`.

then you could use the `classnames` library to conditionally set the active class only like so

```javascript
className={classnames('button-primary', { ['button-active']: selected })}
```